### PR TITLE
Added bubble components and wdio tests 

### LINKED ISSUE #95: Create components and automated test cases for Bubble Graphs
# Feature Request

### Description

As an engineer, I would like to build components and test cases for *bubble graphs*.

### Motivation 
So that we can have improve our dev-site and have automated tests.

## Additional Context / Screenshots

We'll know we're done when the components render on the new dev-site and WDIO test cases are complete.

### PR DESCRIPTION
**Pull request by sdadn**
### Summary

Created bubble components and wdio tests.

Closes #95 

### Deployment Link

https://terra-graphs-deployed-pr-#.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra
@cerner/carbon

### CODE REVIEW

File: packages/terra-graphs/src/components/Bubble/BubbleSingleDataset.jsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  timeout: PropTypes.array,
+};
+
+const BubbleSingleDataset = ({
+  graphConfig, dataset, graphID, timeout,
+}) => {
+  React.useEffect(() => {
+    const graph = Carbon.api.graph(graphConfig);
+    if (!(utils.isUndefined(dataset))) {
+      if (!(utils.isUndefined(timeout))) {
```
Comment (yuderekyu) [inline packages/terra-graphs/src/components/Bubble/BubbleSingleDataset.jsx:33]:
```suggestion
    if (dataset) {
      if (timeout) {
```

File: packages/terra-graphs-docs/src/example-datasets/popup.js [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+const getDate = (d) => new Date(d).toLocaleString();
+const getPairData = (value) =>
+    value ? (checkDate(value) ? getDate(value) : value) : null;
+const checkDate = (date) => date instanceof Date;
+const createItem = (pair, label, value) => {
+    const item = pair.append("g").classed("popup-item", true);
+    item.append("label").classed("popup-label", true).text(label);
+    item.append("g").classed("popup-text", true).text(value);
+};
+const createTrackItem = (pair, label, items) => {
```
Comment (yuderekyu) [inline packages/terra-graphs-docs/src/example-datasets/popup.js:15]: I made a similar comment on a separate graph pull request, but I think converting most of these functions into an object based api will really help clarify what each function argument represents.
```suggestion
const createTrackItem = ({ pair, label, items }) => {

// useage
createTrackItem({ pair: element, label: 'real-world-label', items: dataArr });
```

Comment (sdadn) [reply packages/terra-graphs-docs/src/example-datasets/popup.js:15]: I agree, but I think that's beyond the scope of this issue. The library has some questionable design choices and refactoring those is a separate project that will take more than an iteration. I'd rather avoid making api design changes until we're ready to tackle it as dedicated project due to the spaghetti nature of the implementation and avoid breaking the library.

Comment (yuderekyu) [reply packages/terra-graphs-docs/src/example-datasets/popup.js:15]: Here are some key questions - is this pull request moving old code into the terra-graphs repo, or are these functions brand new on your side? If it's the former then I agree with you, but if it's the latter, I disagree and think it's in our best interest to update the api. 

Writing some context in the pr description would help everyone understand the intent of the pull request. Like "migrating bubble examples from carbon into terra-graphs to render on dev site for integration tests"

Comment (sdadn) [reply packages/terra-graphs-docs/src/example-datasets/popup.js:15]: I apologize for the lack of clarity. These functions and datasets are being migrated from the old dev-site.

**Comment by sdadn** [comment]
@tbiethman @yuderekyu @AshishMotanamGurunadham  I believe I finished updated comments. A few updates to the PR:
1. There was a typo in one of the graph folders (piard results). Moved them to the correct folder.
2. Had to delete and rerun all wdio tests, so you may see some updates for other graphs. There are no changes but git detects them as new files.

**Comment by AshishMotanamGurunadham** [comment]
> @tbiethman @yuderekyu @AshishMotanamGurunadham I believe I finished updated comments. A few updates to the PR:
> 
> 1. There was a typo in one of the graph folders (piard results). Moved them to the correct folder.
> 2. Had to delete and rerun all wdio tests, so you may see some updates for other graphs. There are no changes but git detects them as new files.

May I know the reason why you had to generate new images for all tests? I mean will we get conflicts when merged with main?

And can you also move dev.module.scss file to terra-graphs-docs(Just merge with main). https://github.com/cerner/terra-graphs/tree/main/packages/terra-graphs-docs/src/Css

File: packages/terra-graphs-docs/src/terra-dev-site/doc/example/Graphs/BubbleGraph/ColorBasedGraph.jsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import bubbleDataColorBased from '@cerner/terra-graphs-docs/src/example-datasets/dataObjects/Bubble/bubbleDataColorBased.js';
+
+/*
+Please refer to the documentation below to see graphConfig and data objects.
+*/
+const graphConfig = utils.deepClone(getSimpleAxisData('#colorBasedBubbleGraph'));
+const dataset = utils.deepClone(bubbleDataColorBased);
+
+export default () => (
+  <React.Fragment>
```
Comment (yuderekyu) [inline packages/terra-graphs-docs/src/terra-dev-site/doc/example/Graphs/BubbleGraph/ColorBasedGraph.jsx:16]: Not a big deal, but I just want to point out that React has a shorthand syntax for fragments
```suggestion
  <>
  </>
```

**Review by yuderekyu — APPROVED** [review]

**Comment by sdadn** [comment]
@AshishMotanamGurunadham I have rebased and updated the scss file imports here: 4b40fb5cb0cabecb4192d706b5b807166ea9ed01
There shouldn't be any merge conflicts with main as Github detects the old files as being deleted and new ones being added.

**Review by AshishMotanamGurunadham — APPROVED** [review]

**Review by tbiethman — APPROVED** [review]

**Pull request merged**